### PR TITLE
Group chat

### PR DESCRIPTION
**Pull request by Valentina1133**
Group chat test case automated
Scenario - https://ethstatus.testrail.net/index.php?/cases/view/703014

Screenshot 2023-10-04 at 21 05 31

### CODE REVIEW

File: gui/objects_map/component_names.py [hunk 1]
```diff
@@ -292,3 +292,10 @@
 copy_SyncCodeStatusButton = {"container": statusDesktop_mainWindow_overlay, "objectName": "syncCodeCopyButton", "type": "StatusButton", "visible": True}
 done_SyncCodeStatusButton = {"container": statusDesktop_mainWindow_overlay, "objectName": "syncAnewDeviceNextButton", "type": "StatusButton", "visible": True}
 syncCodeInput_StatusPasswordInput = {"container": statusDesktop_mainWindow_overlay, "id": "syncCodeInput", "type": "StatusPasswordInput", "unnamed": 1, "visible": True}
+
+# Edit group name and image popup
+groupChatEdit_name_TextEdit = {"container": statusDesktop_mainWindow_overlay, "objectName": "groupChatEdit_name", "type": "TextEdit", "visible": True}
+save_changes_StatusButton = {"checkable": False, "container": statusDesktop_mainWindow_overlay, "objectName": "groupChatEdit_save", "text": "Save changes", "type": "StatusButton", "visible": True}
```
Comment (StateOf-Vlado) [inline gui/objects_map/component_names.py:298]: remove text property

File: gui/objects_map/component_names.py [hunk 2]
```diff
@@ -292,3 +292,10 @@
...
+
+# Leave group popup
+leave_StatusButton = {"checkable": False, "container": statusDesktop_mainWindow_overlay, "objectName": "leaveGroupConfirmationDialogLeaveButton", "text": "Leave", "type": "StatusButton", "visible": True}
```
Comment (StateOf-Vlado) [inline gui/objects_map/component_names.py:301]: same here

File: gui/screens/messages.py [hunk 3]
```diff
@@ -115,10 +143,102 @@ def accept_community_invite(self, community: str) -> 'CommunityScreen':
...
+
+    @allure.step('Select contact in the list')
+    def select_contact(self, contact: str):
+        assert driver.waitFor(lambda: contact in self.contacts), f'Contact: {contact} not found in {self.contacts}'
+        self._create_chat_contacts_list.select(contact, 'title')
+
+    @allure.step('Create chat by adding contacts from contact list')
+    def create_chat(self, members):
+        for member in members[0:]:
+            time.sleep(0.2)
```
Comment (StateOf-Vlado) [inline gui/screens/messages.py:167]: strange time.sleep, it is soo small, is it nesessery?

File: gui/screens/messages.py [hunk 4]
```diff
@@ -115,10 +143,102 @@ def accept_community_invite(self, community: str) -> 'CommunityScreen':
...
+        time.sleep(2)
+        self.open_more_options()
+        time.sleep(2)
+        self._leave_group_item.click()
+        return LeaveGroupPopup().wait_until_appears()
+
+    @allure.step('Send message to group chat')
+    def send_message_to_group_chat(self, message: str):
+        self._message_field.type_text(message)
+        driver.nativeType(str('<Return>'))
```
Comment (StateOf-Vlado) [inline gui/screens/messages.py:221]: try to use driver.type(self._message_field.object, '<Return>') 
it is better to send action to a direct object than native

File: tests/test_messaging.py [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            main_window.prepare()
+            profile_popup = main_window.left_panel.open_user_canvas().open_profile_popup()
+            chat_key = profile_popup.chat_key
+            profile_popup.close()
+            main_window.hide()
+
+        with step(f'User {user_one.name}, send contact request to {user_three.name}'):
+            aut_one.attach()
+            main_window.prepare()
+            settings = main_window.left_panel.open_settings()
```
Comment (StateOf-Vlado) [inline tests/test_messaging.py:75]: Looks like we need to join all lines from 75 to 79 in one method - send_contacts_request. what do you think?
In SettingsScreen for example

**Review by StateOf-Vlado — APPROVED** [review]
A few comments added with minor changes
Great job!

File: gui/screens/messages.py (hunk 3 above)
Comment (Valentina1133) [reply gui/screens/messages.py:167]: Yes, I tried without it, but the list content changes after selecting a user so, this small time.sleep needed, so the list updated (it also was like this in old framework)

File: gui/screens/messages.py (hunk 4 above)
Comment (Valentina1133) [reply gui/screens/messages.py]: For some reason it doesn't work here, only with nativeType

**Comment by anastasiyaig** [comment]
@Valentina1133 great job, this is impressive!

**Review by anastasiyaig — APPROVED** [review]

File: gui/screens/messages.py (hunk 4 above)
Comment (StateOf-Vlado) [reply gui/screens/messages.py:221]: '<Return>' - it is a string, you don't need to convert it again to a str()

Comment (StateOf-Vlado) [reply gui/screens/messages.py:221]: driver.nativeType('<Return>')

**Pull request merged**